sympy/functions/special/zeta_functions.py

```diff
@@ -601,5 +622,14 @@ def eval(cls, n, a=None):
                 return S.ComplexInfinity
             elif not n.is_Integer:
                 return S.ComplexInfinity
-            elif n == 0 and a in [None, 1]:
+            elif n is S.Zero and a in [None, 1]:
                 return S.EulerGamma
+
+        if n.is_extended_negative:
+            return S.ComplexInfinity
+
+        if n.is_zero and a in [None, 1]:
+            return S.EulerGamma
```

Comment:
I'd use a tuple for `[None, 1]`. I believe that in CPython it can be embedded as a "compile time constant" in the pyc so a new list isn't created each time the function is called.